PyStemmusScope/save.py

```diff
@@ -0,0 +1,334 @@
+"""PyStemmusScope save module.
+
+Module designed to create a netcdf file following [ALMA
+convention](https://web.lmd.jussieu.fr/~polcher/ALMA/) from csv files following
+[SCOPE format](https://scope-model.readthedocs.io/en/latest/outfiles.html) in
+the output directory.
+
+The file
+[Variables_will_be_in_NetCDF_file.csv](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/utils/csv_to_nc/Variables_will_be_in_NetCDF_file.csv)
+lists required variable names and their attributes based on [`ALMA+CF`
+convention
+table](https://docs.google.com/spreadsheets/d/1CA3aTvI9piXqRqO-3MGrsH1vW-Sd87D8iZXHGrqK42o/edit#gid=2085475627)..
+
+example:
+from PyStemmusScope import StemmusScope
+from PyStemmusScope import save
+
+model = StemmusScope(config_file, exe_file)
+model.setup()
+model.run()
+save.to_netcdf(model.config, "Variables_will_be_in_NetCDF_file.csv")
+"""
+
+import logging
+from pathlib import Path
+from typing import Dict
+from typing import List
+import numpy as np
+import pandas as pd
+import xarray as xr
+from PyStemmusScope import forcing_io
+from . import variable_conversion as vc
+
+
+logger = logging.getLogger(__name__)
+
+
+def _select_forcing_variables(forcing_dict: Dict, forcing_var: str, alma_var: str) -> xr.DataArray:
+    """Select the variable needed by ALMA convention.
+
+    Args:
+        forcing_dict(dict): a dictionary returned by `PyStemmusScope.forcing_io.read_forcing_data()`.
+        forcing_var(str): variable name in forcing dataset.
+        alma_var(str): variable name in ALMA convention.
+
+    Returns:
+        xr.DataArray: a data array which its variable name is alma_name.
+    """
+
+    # select the forcing variable
+    data_array = forcing_dict[forcing_var]
+
+    # rename the variable name to alma_name
+    data_array = data_array.rename(alma_var)
+    return data_array
+
+
+def _resize_data_array(data: xr.DataArray, time_steps: str)-> xr.DataArray:
+    """Resize data based on time_steps.
+
+    Args:
+        data(xr.DataArray): data to be resized.
+        time_steps(str): number of time steps to resize.
+
+    Returns:
+        xr.DataArray: subset of data with the lenght of time equal to time_steps.
+    """
+
+    if time_steps != "NA":
+        time_length = int(time_steps)
+        data = data.isel(time=np.arange(0, time_length))
+
+    return data
+
+
+def _prepare_4d_data(file_name: str, var_name: str, time: List) -> xr.DataArray:
```

Comment:
Right. The util `csv_to_nc` prepare three types of data:

- forcing data that are 3d  (x,y, time)
- model simulations that are 3d  (x, y, time)
- soil temp and mosit simulations that are 4d  (x, y, z, time)

So that was the reason for naming the function like this. But I agree it is not clear. I rename it to `_prepare_soil_data` and also fixed the docstring.